### PR TITLE
[ELF] Move elf::symtab into Ctx

### PR DESCRIPTION
**Pull request by MaskRay**
Remove the global variable `symtab` and add a member variable
(`std::unique_ptr<SymbolTable>`) to `Ctx` instead.

This is one step toward eliminating global states.

### CODE REVIEW

File: lld/ELF/SyntheticSections.cpp [hunk 1]
```diff
@@ -2327,8 +2327,9 @@ SymtabShndxSection::SymtabShndxSection()
 
 void SymtabShndxSection::writeTo(uint8_t *buf) {
   // We write an array of 32 bit values, where each value has 1:1 association
-  // with an entry in .symtab. If the corresponding entry contains SHN_XINDEX,
-  // we need to write actual index, otherwise, we must write SHN_UNDEF(0).
+  // with an entry in .ctx.symtab-> If the corresponding entry contains
```
Comment (smithp35) [inline lld/ELF/SyntheticSections.cpp:2330]: Is this comment change intentional. The code looks to be referring to the .symtab section ctx.in.symTab.

File: lld/ELF/Arch/ARM.cpp [hunk 2]
```diff
@@ -1265,7 +1265,7 @@ void elf::processArmCmseSymbols() {
     return;
   // Only symbols with external linkage end up in symtab, so no need to do
```
Comment (smithp35) [inline lld/ELF/Arch/ARM.cpp:1266]: Could replace symtab with ctx.symtab in the comment.

**Review by smithp35 — APPROVED** [review]
LGTM for the code changes and approach. I've highlighted a couple of places where I think the comments may not line up.

**Comment by MaskRay** [comment]
Thanks for the review and suggestion. Closed as df0864e761107b07e38f5503e0cbee0cebb4c5e8 (forgot `spr land`)

**Pull request closed without merging**